api/src/main/java/org/eclipse/microprofile/context/ThreadContext.java

```diff
@@ -483,8 +485,13 @@ public void foo() {
      * <p>Returns a new <code>CompletableFuture</code> that is completed by the completion of the
      * specified stage.</p>
      *
-     * <p>The new completable future is backed by a ThreadContext rather than a ManagedExecutor.
-     * Therefore, the new stage and all dependent stages created from it, and so forth,
+     * <p>The new completable future will use the same default executor as this ThreadContext,
+     * which can be a ManagedExecutor if this ThreadContext was obtained by {@link ManagedExecutor#getThreadContext()}
+     * or the default executor service as set by
+     * {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)},
```

Comment:
ContextManager is an SPI that is invoked by our implementation rather than application-level interface.  I'd like to have the JavaDoc primarily focus on the application developer here and what makes sense to them.  Maybe it could say:
`or the default executor service if provided by the platform (which can be done via {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)}),`